src/V3Graph.h

```diff
@@ -172,7 +172,106 @@ class V3Graph VL_NOT_FINAL {
 
 //============================================================================
 
+// Holds list of types. For use mainly in compile-time code generation.
+template <typename... TN>
+struct VTypeList {
+    template <typename... UN>
+    constexpr VTypeList<TN..., UN...> operator+(VTypeList<UN...>) const {
+        return {};
+    }
+};
+
+namespace internal_vl {
+
+// Holds one type. Can be safely used as a return or argument type without any potential
+// limitations of held type.
+template <typename T>
+struct VTypeWrapper {
+    using type = T;
+};
+
+template <typename T0, typename... TN>
+static inline constexpr VTypeWrapper<T0> vlTlFront(VTypeList<T0, TN...>) {
+    return {};
+}
+
+}  // namespace internal_vl
+
+// Alias for front type held by type list `TL`.
+template <typename TL>
+using VTypeListFront = typename decltype(internal_vl::vlTlFront(TL{}))::type;
+
+template <typename TL1, typename TL2>
+using VJoinedTypeLists = decltype(TL1{} + TL2{});
+
+namespace internal_vl {
+
+inline static constexpr bool isClassIdOfOneOf(uintptr_t id, VTypeList<>) VL_PURE { return false; }
+
+template <typename Base0, typename... BaseN>
+inline static constexpr bool isClassIdOfOneOf(uintptr_t id, VTypeList<Base0, BaseN...>) VL_PURE {
+    return id == Base0::rttiClassId() || isClassIdOfOneOf(id, VTypeList<BaseN...>{});
+}
+
+#define INTERNAL_VL_RTTI_COMMON_IMPLEMENTATION(ThisClass) \
+private: \
+    struct RttiUniqueTypeForThisClass {}; \
+    static_assert( \
+        std::is_same<RttiUniqueTypeForThisClass, ThisClass::RttiUniqueTypeForThisClass>::value, \
+        "'ThisClass' argument (" #ThisClass ") does not match the class name"); \
+\
+public: \
+    /* Returns unique ID of the class. Useful with `isInstanceOfClassWithId()` method. */ \
+    static uintptr_t rttiClassId() VL_PURE { \
+        /* The only purpose of the following variable is to occupy an unique memory address. */ \
+        /* This address is used as an unique class ID. */ \
+        static char aStaticVariable; \
+        return reinterpret_cast<uintptr_t>(&aStaticVariable); \
+    }
+
+}  // namespace internal_vl
+
+// Call this macro at the beginning of class definition if the class derives from a
+// class with VL_RTTI_IMPLEMENTATION or VL_RTTI_IMPLEMENTATION_BASE calls.
+#define VL_RTTI_IMPLEMENTATION(ThisClass, DirectBaseClass) \
```

Comment:
```suggestion
#define VL_GRAPH_RTTI_IMPL(ThisClass, DirectBaseClass) \
```

Maybe?  I think it should have GRAPH in the name given that defines are global and this is only related to graphs.  Similar other defines.

Reply:
Since the code has been moved to V3Rtti.h, and it is not in any way limited to graph classes, I've renamed it to `VL_RTTI_IMPL`. Is it OK?